src/main/scala/is/hail/methods/Skat.scala

```diff
@@ -0,0 +1,318 @@
+package is.hail.methods
+
+import is.hail.utils._
+import is.hail.variant._
+import is.hail.expr._
+import is.hail.keytable.KeyTable
+import is.hail.stats.{RegressionUtils, SkatModel}
+import is.hail.annotations.Annotation
+import breeze.linalg._
+import org.apache.spark.rdd.RDD
+import org.apache.spark.sql.Row
+
+object SkatStat {
+  val schema = TStruct(
+    ("q", TFloat64),
+    ("pval", TFloat64),
+    ("fault", TInt32))
+}
+
+case class SkatStat(q: Double, pval: Double, fault: Int) {
+  def toAnnotation = Annotation(q, pval, fault)
+}
+
+case class SkatTuple[T <: Vector[Double]](q: Double, xw: T, qtxw: DenseVector[Double])
```

Comment:
This should either have a better name, or it's not worth the extra complexity above using a Tuple3 `(Double, T <: Vector[Double], DenseVector[Double])`